app/feature/feature-delete-account/src/main/kotlin/com/hedvig/android/feature/deleteaccount/DeleteAccountPresenter.kt

```diff
@@ -0,0 +1,113 @@
+package com.hedvig.android.feature.deleteaccount
+
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableIntStateOf
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.runtime.snapshots.Snapshot
+import com.hedvig.android.core.common.safeCast
+import com.hedvig.android.feature.deleteaccount.data.DeleteAccountState
+import com.hedvig.android.feature.deleteaccount.data.DeleteAccountStateUseCase
+import com.hedvig.android.feature.deleteaccount.data.RequestAccountDeletionUseCase
+import com.hedvig.android.molecule.public.MoleculePresenter
+import com.hedvig.android.molecule.public.MoleculePresenterScope
+import kotlinx.coroutines.flow.collectLatest
+
+internal class DeleteAccountPresenter(
+  private val requestAccountDeletionUseCase: RequestAccountDeletionUseCase,
+  private val deleteAccountStateUseCase: DeleteAccountStateUseCase,
+) : MoleculePresenter<DeleteAccountEvent, DeleteAccountUiState> {
+  @Composable
+  override fun MoleculePresenterScope<DeleteAccountEvent>.present(
+    lastState: DeleteAccountUiState,
+  ): DeleteAccountUiState {
+    var loadIteration by remember { mutableIntStateOf(0) }
+    var isPerformingDeletion by remember { mutableStateOf(false) }
+    var failedToPerformAccountDeletion by remember {
+      mutableStateOf(lastState.safeCast<DeleteAccountUiState.Success.CanDelete>()?.failedToPerformDeletion ?: false)
+    }
+    var deleteAccountState: DeleteAccountState? by remember {
+      mutableStateOf(lastState.toDeleteAccountStateResult())
+    }
+
+    LaunchedEffect(loadIteration) {
+      failedToPerformAccountDeletion = false
+      if (deleteAccountState == DeleteAccountState.NetworkError) {
+        deleteAccountState = null
+      }
+      deleteAccountStateUseCase.invoke().collectLatest { stateResult ->
+        deleteAccountState = stateResult
+      }
+    }
+
+    CollectEvents { event ->
+      when (event) {
+        DeleteAccountEvent.InitiateAccountDeletion -> isPerformingDeletion = true
+        DeleteAccountEvent.RetryLoading -> loadIteration++
+      }
+    }
+
+    LaunchedEffect(isPerformingDeletion) {
+      if (!isPerformingDeletion) return@LaunchedEffect
+      if (deleteAccountState !is DeleteAccountState.CanDelete) {
+        isPerformingDeletion = false
+        return@LaunchedEffect
+      }
+      val mutationResult = requestAccountDeletionUseCase.invoke()
+      Snapshot.withMutableSnapshot {
+        if (mutationResult.isLeft()) {
+          failedToPerformAccountDeletion = true
```

Comment:
Are we fine with showing a generic error here? Are we logging the type of error somewhere?

Reply:
My intention with this was that if the mutation itself fails, we show an error state which they can click retry on and click the button again.
It was also partly because we got no design for this, nor error case copy so I tried to make do with what was already available there.

But yes, let's try to log something here.

This is one of those places where I'd love to be able to log an error but not log again (since we do it in our `DatadogInterceptor`) if the error was network related.

It's one of those things I will fix when we bump to apollo-kotlin 4.x when that is out of beta.

So I will log as a Warning instead of an error to avoid pinging on-call people when normal network errors happen

Reply:
Addressed:
https://github.com/HedvigInsurance/android/commit/9a0fb2db79b4cd08b58c2c6674d87722c391a863